numpy/f2py/crackfortran.py

```diff
@@ -1933,9 +1929,7 @@ def setattrspec(decl, attr, force=0):
         decl['attrspec'].append(attr)
     if attr in decl['attrspec']:
         return decl
-    if attr == 'static' and 'automatic' not in decl['attrspec']:
-        decl['attrspec'].append(attr)
-    elif attr == 'automatic' and 'static' not in decl['attrspec']:
+    if attr == 'static' and 'automatic' not in decl['attrspec'] or attr == 'automatic' and 'static' not in decl['attrspec']:
```

Comment:
I think the original is more readable.

Reply:
Agreed. Perhaps temporary variables with meaningful names would be even easier to read.